cpp/order/Steinhardt.cc

```diff
@@ -0,0 +1,495 @@
+			iter != sph_eval.end(); ++iter)
+			{
+			Ylm[j] = *iter;
+			++j;
+			}
+		}
+    }
+
+void Steinhardt::reallocateArrays(unsigned int Np)
+	{
+		m_Qlmi = std::shared_ptr<complex<float> >(new complex<float> [(2*m_l+1)*m_Np], std::default_delete<complex<float>[]>());
+		m_Qli = std::shared_ptr<float>(new float[m_Np], std::default_delete<float[]>());
+		m_Qlm = std::shared_ptr<complex<float> >(new complex<float>[2*m_l+1], std::default_delete<complex<float>[]>());
+		if (m_average)
+		{
+			m_AveQlmi = std::shared_ptr<complex<float> >(new complex<float> [(2*m_l+1)*Np], std::default_delete<complex<float>[]>());
+			m_QliAve = std::shared_ptr<float>(new float[Np], std::default_delete<float[]>());
+			m_AveQlm = std::shared_ptr<complex<float> >(new complex<float> [(2*m_l+1)], std::default_delete<complex<float>[]>());
+		}
+		if (m_Wl)
+		{
+			if (m_average && m_norm)
+			{
+				m_WliAveNorm = std::shared_ptr<complex<float> >(new complex<float>[m_Np], std::default_delete<complex<float>[]>());
+			}
+			else if (m_average)
+			{
+				m_WliAve = std::shared_ptr<complex<float> >(new complex<float>[Np], std::default_delete<complex<float>[]>());
+			}
+			else if (m_norm)
+			{
+				m_WliNorm = std::shared_ptr<complex<float> >(new complex<float>[m_Np], std::default_delete<complex<float>[]>());
+			}
+			else
+			{
+				m_Wli = std::shared_ptr<complex<float> >(new complex<float>[Np], std::default_delete<complex<float>[]>());
+			}
+		}
+		else
+		{
+			if (m_average && m_norm)
+			{
+				m_QliAveNorm = std::shared_ptr<float>(new float[Np], std::default_delete<float[]>());
+			}
+			else if (m_norm)
+			{
+				m_QliNorm = std::shared_ptr<float>(new float[Np], std::default_delete<float[]>());
+			}
+		}
+	}	
+
+void Steinhardt::compute(const locality::NeighborList *nlist, const vec3<float> *points, unsigned int Np)
+{
+	// Conditional reinitialize arrays if size differs from previous call.
+	if (m_Np != Np)
+	{
+		m_Np = Np;
+		Steinhardt::reallocateArrays(Np);
+	}
+	// Computes the base Q required for each specialized order parameter
+	Steinhardt::baseCompute(nlist, points);
+	
+	if (m_average){
+		Steinhardt::computeAve(nlist, points);
+	}
+	if (m_Wl){
+		if (m_average && m_norm){
+			Steinhardt::computeAveNormWl();
+		}
+		else if (m_norm){
+			Steinhardt::computeNormWl();
+		}
+		else if (m_average){
+			Steinhardt::computeAveWl();
+		}
+		else {
+			Steinhardt::computeWl();
+		}
+	}
+	else {
+		if (m_average && m_norm){
+			Steinhardt::computeAveNorm();
+		}
+		else if (m_norm){
+			Steinhardt::computeNorm();
+		}
+	}
+}
+
+void Steinhardt::baseCompute(const locality::NeighborList *nlist, const vec3<float> *points)
+    {
+    nlist->validate(m_Np, m_Np);
+
+    memset((void*) m_Qlmi.get(), 0, sizeof(complex<float>)*(2*m_l+1)*m_Np);
+    memset((void*) m_Qli.get(), 0, sizeof(float)*m_Np);
+
+    parallel_for(tbb::blocked_range<size_t>(0, m_Np),
+        [=] (const blocked_range<size_t>& r)
+        {
+	// May not be necessary check
```

Comment:
What check isn't necessary? Is there something we can remove here, or is this a copied comment from old code?

Reply:
Copied from old code